### PR TITLE
Multiversion doc

### PR DESCRIPTION
**Pull request by hndgzkn**
Changes sphinx theme from sphinx-bootstap to sphinx-pydata-theme to display documentation for every release.

This requires some changes in
*  https://github.com/alphacsc/alphacsc.github.io repo:
    - a version.txt file in root directory of the project that contains the latest stable version
    - a dev directory that contains documentation for development version
    - a stable directory that contains documentation for the latest release
    - versionwarning.js file that displays a warning and link to stable version on versions other than stable 

* release process:
   - before each release doc/_static/switcher.json file should be updated with new release version.
   
Documentation generation process:
- For each push/pr merge to master branch or release candidate tag with 'rc' generated documentation in pushed to dev directory under documentation repo.
- For each release generated documentation is pushed to stable directory.

### CODE REVIEW

File: .github/workflows/deploy_ghpages.yml [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Cache documentation
+        if: ${{ github.ref_type }} != 'tag'
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-documentation
+        with:
+          path: doc/_build/
```
Comment (tomMoral) [inline .github/workflows/deploy_ghpages.yml:26]: I think you mainly need to cache the example no?
```suggestion
          path: |
            doc/_build/
            doc/auto_examples
```

**Review by tomMoral — APPROVED** [review]
LGTM. If my suggestion fixes the cache, lets use it, else, let's move on and fix the cache latter on.

File: .github/workflows/deploy_ghpages.yml (hunk 1 above)
Comment (hndgzkn) [reply .github/workflows/deploy_ghpages.yml:26]: Yes, actually I was thinking that sphinx used _build/doctree directory, probably your change will work then. Thanks!

Shall I update the [documentation repo](https://github.com/alphacsc/alphacsc.github.io) then?

Comment (tomMoral) [reply .github/workflows/deploy_ghpages.yml:26]: If this is working, feel free to change the doc and merge :)

Comment (hndgzkn) [reply .github/workflows/deploy_ghpages.yml:26]: Unfortunately cache still does not work :(

**Comment by tomMoral** [comment]
The cache is now working! 🎉

**Review by agramfort — APPROVED** [review]
Nice ! Merge if happy @hndgzkn

**Comment by hndgzkn** [comment]
> The cache is now working! tada

Yes, it is working :) 

But there is a problem with deploy job. I try to understand the reason. There are recent changes in [Push action](https://github.com/ad-m/github-push-action) that we use. I try to understand if it is due to these changes.

**Comment by hndgzkn** [comment]
Probably not as we pinned to version 0.6.0 :(

**Comment by hndgzkn** [comment]
Ok. It was due to incorrect if condition in deploy job. I fixed it. I am merging

**Pull request merged**